_posts/2023-11-29-java-17.adoc

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: 'Java 17 will be the minimum version for Quarkus 3.7'
+date: 2023-11-29
+tags: announcement
+synopsis: Quarkus 3.7 will move Quarkus to use Java 17 as the minimal JDK
+author: maxandersen
+---
+
+:imagesdir: /assets/images/posts/
+ifdef::env-github,env-browser,env-vscode[:imagesdir: ../assets/images/posts/]
+
+Today we are anouncing that Quarkus 3.7 will move Quarkus to use Java 17 as the minimum JDK.
+
+When we started Quarkus 3.0, https://quarkus.io/blog/road-to-quarkus-3/#java-target[we stayed on Java 11]. The feedback we got was that moving to Jakarta EE 10 was for many easier than to also move the underlyng JDK. We wanted to give the ecosystem time to catch up and make sure we had a good story for users that needed to stay on Java 11. 
+
+In 3.0 release, we supported Java 11 for Quarkus core, and we https://quarkus.io/blog/quarkus-3-0-final-released/#java-11-deprecated[marked Java 11 as deprecated]. It should be noted that parts of the platform, especially the Camel Extensions for Quarkus, already require Java 17.
+
+Today we are taking the step to move Quarkus core to Java 17 as the minimum JDK for Quarkus 3.7 and onwards.
+
+This means that Quarkus 3.7 will require Java 17 to build and run all Quarkus applications.
+
+Below are some of the reasons for this decision and what it means for you as a Quarkus user.
+
+## Everyone is using Java 11 - you cannot do this!
+
+We know that many users are still using Java 11 and we are not forcing you to move to Java 17. We are just making it the minimum version for Quarkus 3.7 and onwards.
+
+That said - the https://quarkus.io/usage/[usage data development builds] in the last 30 days shows less than 8% using Java 11 for Quarkus Development. 
+
+image::javaversions-used-by-32plus.png[]
+
+We expect the Java 11 number to be higher in production but the trend is clear - Java 11 usage is low and declining.
+
+NOTE: If you like to help us get more accurate data for future, then please consider contributring anonymous build time data - see https://quarkus.io/usage/[quarkus.io/usage] for details.
+
+## I cannot move to Java 17 yet, what should I do?
+
+If you cannot move to Java 17 yet, then you can continue to use Quarkus 3.6, but we recommend you consider to use the Quarkus 3.2 LTS stream as that is the version that will get updates and fixes for the longest. 
+
+Be mindful though that Java 11 itself is rapidly heading towards going out of full support. Example Red Hat's full https://access.redhat.com/articles/1299013#OpenJDK_Life_Cycle[Java 11 support ends in October 2024] and https://www.oracle.com/java/technologies/java-se-support-roadmap.html[Oracle ended Premier support in September 2023]. Thus we highly recommend you start planning your migration to Java 17 or even Java 21 as soon as possible.
```

Comment:
@tqvarnst can you confirm those levels of red hat full support vs oracle premium support are comparable measures?

Reply:
thomas confirmed offline these are not necessarily comparable but we are also just calling out those levels of support has or is ending.